src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -563,7 +563,7 @@ private static boolean isPrecededByJavadoc(DetailAST token) {
         boolean result = false;
         final DetailAST previous = token.getPreviousSibling();
         if (previous.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
-                && JavadocUtils.isJavadocComment(previous)) {
+                && JavadocUtils.isJavadocComment(previous.getFirstChild().getText())) {
```

Comment:
> I created the PR anyways thinking that we can discuss this in PR

If you want to discuss something, please mention it in first post otherwise we may overlook something in review.

Correct me if I am wrong, but the UT and Input you pointed to isn't even a javadoc. It only has 1 `*` instead of 2 so it shouldn't even pass as a javadoc in a wrong position.